### PR TITLE
DOC: Prepare release notes for 3.9

### PR DESCRIPTION
**Pull request by QuLogic**
## PR summary

This consolidates all the individual notes into single file for the release. A few tweaks were made to the text to flow better.

## PR checklist

- [n/a] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [n/a] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [x] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [x] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by QuLogic** [comment]
I think I might still have to add a few more images to the What's new page.

File: doc/users/prev_whats_new/whats_new_3.9.0.rst [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+Setting the parameter *side* to 'low' or 'high' allows to only plot one half of the
+violin plot.
+
+``axhline`` and ``axhspan`` on polar axes
+-----------------------------------------
+
+... now draw circles and circular arcs (`~.Axes.axhline`) or annuli and wedges
+(`~.Axes.axhspan`).
+
+subplot titles can now be automatically aligned
```
Comment (oscargus) [inline doc/users/prev_whats_new/whats_new_3.9.0.rst:137]:
```suggestion
Subplot titles can now be automatically aligned
```

File: doc/users/prev_whats_new/whats_new_3.9.0.rst [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+               boxprops={'facecolor': 'lightblue'})
+
+    ax.legend()
+
+Percent sign in pie labels auto-escaped with ``usetex=True``
+------------------------------------------------------------
+
+It is common, with `.Axes.pie`, to specify labels that include a percent sign
+(``%``), which denotes a comment for LaTeX. When enabling LaTeX with
+:rc:`text.usetex` or passing ``textprops={"usetex": True}``, this would cause
```
Comment (oscargus) [inline doc/users/prev_whats_new/whats_new_3.9.0.rst:89]:
```suggestion
:rc:`text.usetex` or passing ``textprops={"usetex": True}``, this used to cause
```

File: doc/users/prev_whats_new/whats_new_3.9.0.rst [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+
+    ani = animation.FuncAnimation(fig, update, frames=15, interval=90, blit=False)
+
+    plt.show()
+
+NonUniformImage now has mouseover support
+-----------------------------------------
+
+When mousing over a `~matplotlib.image.NonUniformImage` the data values are now
```
Comment (oscargus) [inline doc/users/prev_whats_new/whats_new_3.9.0.rst:350]:
```suggestion
When mousing over a `~matplotlib.image.NonUniformImage`, the data values are now
```

File: doc/users/prev_whats_new/whats_new_3.9.0.rst [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+``StrMethodFormatter`` now respects ``axes.unicode_minus``
+----------------------------------------------------------
+
+When formatting negative values, `.StrMethodFormatter` will now use unicode minus signs
+if :rc:`axes.unicode_minus` is set.
+
+Figure, Axes, and Legend Layout
+===============================
+
+Subfigures have now controllable zorders
```
Comment (oscargus) [inline doc/users/prev_whats_new/whats_new_3.9.0.rst:165]:
```suggestion
Subfigures now have controllable zorders
```

**Review by oscargus — COMMENTED** [review]
Some minor corrections/improvements(?).

File: doc/api/prev_api_changes/api_changes_3.9.0/behaviour.rst [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+`~matplotlib.axes.Axes.boxplot` and `~matplotlib.cbook.boxplot_stats` now ignore any
+masked points in the input data.
+
+Default behavior of ``hexbin`` with *C* provided requires at least 1 point
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The behavior changed in 3.8.0 to be inclusive of *mincnt*. However that resulted in
+errors or warnings with some reduction functions, so now the default is to require at
+least 1 point to call the reduction function. This effectively restores the default
+behavior to match that of Matplotlib 3.7 and before.
```
Comment (ksunden) [inline doc/api/prev_api_changes/api_changes_3.9.0/behaviour.rst:36]:
```suggestion
```

Included in 3.8.1, apparently the note not removed?

File: doc/api/prev_api_changes/api_changes_3.9.0/deprecations.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
+Deprecations
+------------
+
+``contour`` deprecations reverted
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``contour.allsegs``, ``contour.allkinds``, and ``contour.find_nearest_contour`` are no
+longer marked for deprecation.
+
```
Comment (ksunden) [inline doc/api/prev_api_changes/api_changes_3.9.0/deprecations.rst:9]:
```suggestion
```

Also from 3.8.1

File: doc/users/prev_whats_new/whats_new_3.9.0.rst [hunk 7]
```diff
@@ -119,21 +116,86 @@ Add option to plot only one half of violin plot
...
 -----------------------------------------------
 
 Subplot axes titles can be misaligned vertically if tick labels or xlabels are placed at
-the top of one subplot. The new method on the `.Figure` class: `.Figure.align_titles`
+the top of one subplot. The new `~.Figure.align_titles` method on the `.Figure` class
 will now align the titles vertically.
 
+.. plot::
+    :include-source:
+    :alt: A figure with two Axes side-by-side, the second of which with ticks on top. The Axes titles and x-labels ppear unaligned with each other due to these ticks.
```
Comment (ksunden) [inline doc/users/prev_whats_new/whats_new_3.9.0.rst:165]: ppear -> appear

```suggestion
    :alt: A figure with two Axes side-by-side, the second of which with ticks on top. The Axes titles and x-labels appear unaligned with each other due to these ticks.
```

**Pull request merged**